codegen/smithy-ruby-codegen-test-utils/middleware/test_middleware.rb

```diff
@@ -15,4 +15,48 @@ def call(input, context)
       output
     end
   end
+
+  # Middleware used to test relative middleware ordering -
+  # this is for 'BEFORE' type testing
+  class BeforeMiddleware
+    def initialize(app)
+      @app = app
+    end
+
+    def call(input, context)
+      context.metadata[:before_middleware] = true
+      @app.call(input, context)
+    end
+  end
+
+  # Middleware used to test relative middleware ordering -
+  # checks the 'BEFORE' middleware and an optional case
+  # when a relative middleware is not required
+  class MidMiddleware
+    def initialize(app, verify_before_middleware:)
```

Comment:
This is a little confusing to have the MidMiddlware take a verify_before config (I know it gets used to verify the Before middleware, but from the middleware POV, its only associated with the Mid Middleware.  So maybe something like `verify_in_mid`?

Reply:
I think you can remove all of these configs and simply expect certain middleware to exist in app. It's an "onion" of next middleware.

Reply:
Thanks to y'all - getting rid of the messy configs on these middlewares! 😃